### PR TITLE
ENT-4666: Fixed issue with CourseEnrollment receiver when learner has multiple enterprises

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
…enterprises

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise/signals.py [hunk 1]
```diff
@@ -329,9 +328,8 @@ def create_enterprise_enrollment_receiver(sender, instance, **kwargs):     # pyl
     """
     if kwargs.get('created') and instance.user:
         user_id = instance.user.id
-        try:
-            ecu = EnterpriseCustomerUser.objects.get(user_id=user_id)
-        except ObjectDoesNotExist:
+        ecu = EnterpriseCustomerUser.objects.filter(user_id=user_id, active=True)
```
Comment (HammadAhmadWaqas) [inline enterprise/signals.py:331]: `ecu` is here a `queryset`, and later on we are using it  as `ecu .id` at [line](https://github.com/edx/edx-enterprise/pull/1284/files#diff-b4b6533b246e5794f1a7051c1bc5b1848746e918dad92f5ea5c6ad09ada2fdfdR342). I think code will break here.

**Review by HammadAhmadWaqas — CHANGES_REQUESTED** [review]

**Comment by HammadAhmadWaqas** [comment]
it would be great if can add test for it.

File: enterprise/signals.py (hunk 1 above)
Comment (irfanuddinahmad) [reply enterprise/signals.py:331]: Good catch! ... fixed

**Comment by irfanuddinahmad** [comment]
> it would be great if can add test for it.

Test is already present at https://github.com/edx/edx-enterprise/blob/f14a13ed9ef0dfe36b4ef8e5b2f0d47be996f1ea/tests/test_enterprise/test_signals.py#L698

**Review by zamanafzal — APPROVED** [review]

**Review by HammadAhmadWaqas — APPROVED** [review]
LGTM

**Comment by edx-abolger** [comment]
@irfanuddinahmad, I see that there was a bug, but the solution here seems like it could create an Enterprise Course Enrollment for the WRONG enterprise if the wrong EnterpriseCustomer is chosen?  I think this may be a bigger bug than how you fixed it: what if the EnterpriseCourseEnrollment to create should actually be for the other course? (@iloveagent57 , is there an architectural workaround here? It seems like we might actually be missing the information in the signal that would get the right Enterprise selected, which would have to be passed in somehow from a browser session?)

**Pull request merged**